### PR TITLE
Detect Circuit Playground Type and Firmware Version

### PR DESCRIPTION
**Pull request by epeach**
Corresponds with this ticket: https://codedotorg.atlassian.net/browse/STAR-494
This PR is just to detect this information, but not do anything with it, at the moment.

### CODE REVIEW

**Review by ajpal — APPROVED** [review]
lgtm!

**Comment by joshlory** [comment]
Worth adding these lines before "Can talk to the firmware"?

```js
.then(() => setupChecker.detectBoardType())
.then(() => setupChecker.detectFirmwareVersion())
```

Then we're actually exercising the new code path, even if there's no user-visible change yet.

**Comment by joshlory** [comment]
Ah sorry, `this.detectFirmwareVersion(playground);` would need to go in `connectToFirmware()`.

File: apps/src/lib/kits/maker/CircuitPlaygroundBoard.js [hunk 1]
```diff
@@ -81,6 +98,9 @@ export default class CircuitPlaygroundBoard extends EventEmitter {
       const playground = CircuitPlaygroundBoard.makePlaygroundTransport(
         serialPort
       );
+      if (experiments.isEnabled('detect-board')) {
+        this.detectFirmwareVersion(playground);
+      }
```
Comment (joshlory) [inline apps/src/lib/kits/maker/CircuitPlaygroundBoard.js]: Does this need to happen inside the "board ready" block below?

File: apps/src/lib/kits/maker/CircuitPlaygroundBoard.js [hunk 2]
```diff
@@ -86,13 +103,51 @@ export default class CircuitPlaygroundBoard extends EventEmitter {
...
+          playground.firmware.version.minor
+      );
+    });
+  }
+
+  /**
+   * Detects the type of board plugged into the serial port
+   */
+  detectBoardType() {
+    return findPortWithViableDevice().then(port => {
```
Comment (joshlory) [inline apps/src/lib/kits/maker/CircuitPlaygroundBoard.js:134]: Do we need to re-list the devices here, or can we get it from the promise chain?

**Review by joshlory — APPROVED** [review]
🎉Board version detection!

**Pull request merged**